src/odemis/dataio/tiff.py

```diff
@@ -1833,6 +1837,46 @@ def _saveAsMultiTiffLT(filename, ldata, thumbnail, compressed=True, multiple_fil
             write_image(f, data[i], write_rgb=write_rgb, compression=c, pyramid=pyramid)
 
 
+def extract_imagej_metadata(ldata) -> str:
+    """
+    Create ImageJ compatible metadata that is added to the ImageDescription identifier for the tiff files.
+    :param ldata: (list of DataArray) list of 2D data of int or float. Should have at least one array
+    :return (str): metadata compatible with ImageJ
+    """
+    num_channels = 0
+
+    # Check the CTZYX dimensions of the first DataArray
+    size = ldata[0].shape
+    res = (1,) * (5 - len(size)) + size
+
+    # Add to "channels" if other DataArray(s) have the same TZYX dimension.
+    for data in ldata:
+        next_shape = data.shape
+        next_res = (1,) * (5 - len(next_shape)) + next_shape
+        if next_res == res:
+            if len(data.shape) == 5:
```

Comment:
I think you should just use `next_res[-5]`, then no need to check that the shape is big enough.